File: frontend/src/types/category.ts
export interface Category {
  categoryId: number;
  parentId: number | null;
  categoryName: string;
  categoryLevel: number;
  description: string | null;
  createdAt: string;
}

export interface CategoryTree {
  categoryId: number;
  categoryName: string;
  categoryLevel: number;
  description: string | null;
  children: CategoryTree[];
}

export interface CategoryCreateRequest {
  parentId?: number;
  categoryName: string;
  categoryLevel: number;
  description?: string;
}

export interface CategoryUpdateRequest {
  categoryName: string;
  description?: string;
}

File: mobile/lib/features/asset/data/models/category.dart
class Category {
  final int categoryId;
  final int? parentId;
  final String categoryName;
  final int categoryLevel;
  final String? description;
  final DateTime createdAt;

  const Category({
    required this.categoryId,
    this.parentId,
    required this.categoryName,
    required this.categoryLevel,
    this.description,
    required this.createdAt,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      categoryId: json['categoryId'] as int,
      parentId: json['parentId'] as int?,
      categoryName: json['categoryName'] as String,
      categoryLevel: json['categoryLevel'] as int,
      description: json['description'] as String?,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}
